manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -176,6 +176,17 @@ func (tr *TaskReaper) tick() {
 
 	defer func() {
 		tr.cleanup = nil
+
+		// Clean out the dirty set at the end of this tick iteration.
+		// When tick() finishes, the tasks in the slot were either cleaned up,
+		// or it was skipped because it didn't meet the criteria for cleaning.
+		// Either way, we can discard the dirty set because future events on that slot
+		// will cause the task to be readded to the dirty set at that point.
+		//
+		// TODO(anshul) The only edge case is when there are more than one running tasks present
+		// for a given slot. In that case, the slot may remain uncleaned until there is a task
+		// restart for that slot.
+		tr.dirty = nil
```

Comment:
Is `nil` even a valid value for `tr.dirty`? Won't this just make it segfault on the next `EventCreateTask`?

Reply:
maybe change it to `tr.dirty = make(map[orchestrator.SlotTuple]struct{})`